multidb/tests.py

```diff
@@ -201,3 +202,43 @@ def test_context_manager_exception(self):
                 assert this_thread_is_pinned()
                 raise ValueError
         assert not this_thread_is_pinned()
+
+    def test_multithreaded_unpinning(self):
```

Comment:
This test is some what complex for such a simple change. It may be possible to simplify this test, but I don't have any ideas.

I wanted to avoid triggering the bug in a contrived way with specially constructed calls to `pin_this_thread` and `un pin_this_thread`, so I went the thread/lock route.